File: Lab_Lista11/Zadanie3/Debug.java
package Lab_Lista11.Zadanie3;

import java.lang.reflect.Field;

public class Debug
{
    public static void fields(Object obj) throws NoSuchFieldException, IllegalAccessException
    {
        Field [] fields = obj.getClass().getDeclaredFields();

        for(Field field : fields)
        {
            field.setAccessible(true);
            System.out.println("Pole: " + field.getName() + "=> " + field.getType().toString() + ", " + field.get(obj));
        }
    }
}

File: Lab_Lista11/Zadanie2/Debug.scala
package Lab_Lista11.Zadanie2

trait Debug {
  def debugVars(): Unit = {
    val fields = getClass.getDeclaredFields

    for (field <- fields) {
      field.setAccessible(true)
      println("Pole: " + field.getName + " => " + field.getType + ", " + field.get(this))
    }
  }
}
